File: backend/src/main/java/t/lab/guide/dto/auth/RegistrationResponse.java
package t.lab.guide.dto.auth;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import t.lab.guide.dto.user.UserResponse;

@Schema(description = "Ответ при регистрации нового пользователя, содержит пару токенов и информацию о пользователе")
@Builder
public record RegistrationResponse(
        @Schema(description = "Пара access + refresh токенов")
        TokenPairResponse tokens,
        @Schema(description = "Информация о пользователе")
        UserResponse user
){}

File: backend/src/main/kotlin/t/lab/guide/dto/auth/RegistrationResponse.kt
package t.lab.guide.dto.auth

import io.swagger.v3.oas.annotations.media.Schema
import t.lab.guide.dto.user.UserResponse

@Schema(description = "Ответ при регистрации нового пользователя, содержит пару токенов и информацию о пользователе")
data class RegistrationResponse(
    @Schema(description = "Пара access + refresh токенов")
    val tokens: TokenPairResponse,
    @Schema(description = "Информация о пользователе")
    val user: UserResponse,
)
